### PR TITLE
Translate state and lifecycle to Bahasa

### PR DESCRIPTION
**Pull request by amiruddinid**
I have been done translating this section, please kindly review

I keep some of word in english that maybe reference to React function or method 
like "Mount" & "Unmounting" for componentWillMount() & componentWillUnmount() lifecycle methods

### CODE REVIEW

**Comment by gedeagas** [comment]
Awesome, thank you for contributing. I will check your work tommorow

File: content/docs/state-and-lifecycle.md [hunk 1]
```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
...
 ```js
-// Wrong
+// Salah
 this.setState({
   counter: this.state.counter + this.props.increment,
 });
 ```
 
-To fix it, use a second form of `setState()` that accepts a function rather than an object. That function will receive the previous state as the first argument, and the props at the time the update is applied as the second argument:
+Untuk memperbaikinya, pakai bentuk kedua dari `setState()` yang menerima fungsi rather daripada sebuah object. Fungsi itu akan menerima _state_ sebelumnya sebagai argumen pertama, dan _props_ pada waktu itu pembaruanya di terapkan ke argumen kedua:
```
Comment (rizafahmi) [inline content/docs/state-and-lifecycle.md]: * ~~rather~~
* object -> _object_ atau objek

File: content/docs/state-and-lifecycle.md [hunk 2]
```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
...
 ```js
 // Correct
 this.setState((state, props) => ({
   counter: state.counter + props.increment
 }));
-```
+``` 
 
-We used an [arrow function](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) above, but it also works with regular functions:
+Kita menggunakan [_arrow function_](https://developer.mozilla.org/id/docs/Web/JavaScript/Reference/Functions/Arrow_functions) diatas, tapi itu juga berkerja dengan fungsi biasa:
```
Comment (rizafahmi) [inline content/docs/state-and-lifecycle.md]: "tapi itu juga berkerja dengan fungsi biasa:" -> "tetapi juga bisa menggunakan fungsi biasa:"

File: content/docs/state-and-lifecycle.md [hunk 3]
```diff
@@ -244,21 +244,21 @@ The `componentDidMount()` method runs after the component output has been render
   }
 ```
 
-Note how we save the timer ID right on `this`.
+Perhatikan bagaimana kami menyimpan _ID_ pengatur _timer_ langsung di `this`.
 
-While `this.props` is set up by React itself and `this.state` has a special meaning, you are free to add additional fields to the class manually if you need to store something that doesn’t participate in the data flow (like a timer ID).
+Ketika `this.props` diatur oleh React sendiri dan `this.state` punya arti spesial, Anda dapat dengan bebas untuk menambah bidang tambahan di kelas secara manual jika Anda butuh untuk menyimpan sesuatu yang tidak ikut berpartisipasi di alur data (seperti _ID timer_).
```
[the remark this answers is not in the document]
Comment (amiruddinid) [reply content/docs/state-and-lifecycle.md]: @rizafahmi untuk bagian ini yang di ganti mulai dari `this.state` kah ?

[the remark this answers is not in the document]
Comment (rizafahmi) [reply content/docs/state-and-lifecycle.md]: bidang tambahan aja diganti jadi _field_ tambahan

[the remark this answers is not in the document]
Comment (amiruddinid) [reply content/docs/state-and-lifecycle.md]: okay siap

**Comment by amiruddinid** [comment]
The requested modifications have been completed., please kindly check @rizafahmi @gedeagas,
thank you

**Comment by rizafahmi** [comment]
All good. Well done @amiruddinid. It's all yours @gedeagas !!

**Pull request merged**